File: module01/dec_to_base.go
package module01

import "fmt"

// DecToBase will return a string representing
// the provided decimal number in the provided base.
// This is limited to bases 2-16 for simplicity.
//
// Eg:
//
//   DecToBase(14, 16) => "E"
//   DecToBase(14, 2) => "1110"
//
func DecToBase(dec, base int) string {
	var result string
	for dec > 0 {
		rem := dec % base
		result = fmt.Sprintf("%X", rem) + result
		dec = dec / base
	}
	return result
}

File: module01/dec_to_base.py
import sys

def dec_to_base(num, base) :
    result = ""
    while num > 0 :
      rem = num%base
      result = "%X"%rem +result
      num = num // base
    return result

if __name__ == "__main__":
    num, base = sys.argv[1], sys.argv[2]
    output = dec_to_base(int(num),int(base))
    print(output)